datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -739,7 +740,9 @@ impl OptimizerRule for PushDownFilter {
 
                             (field.qualified_name(), expr)
                         })
-                        .partition(|(_, value)| is_volatile_expression(value));
+                        .partition(|(_, value)| {
+                            is_volatile_expression(value).unwrap_or(true)
```

Comment:
why use `unwrap_or(true)`?

Reply:
If we don't know whether the scalar function is volatile, default set it to a volatile function.